### PR TITLE
feat: Support PeriodicTokenVesting Smart Contract

### LINKED ISSUE #31: Support PeriodicTokenVesting contract
Support the new vesting contract.

Update this message with a relevant one for the new one

Update this message as well

This sidebar as well

The graphic should look something more like a staircase.

### PR DESCRIPTION
**Pull request by fzavalia**
Closes #31 

By merging this PR, users will be able to use this dashboard to view vestings deployed with the [PeriodicTokenVesting](https://github.com/decentraland/vestings-builder/blob/feat/token-vesting-v2/contracts/PeriodicTokenVesting.sol) contract.

Users can still use the same mechanism to view the smart contract by just writing the contract's address in the URL `/#/0x4deb9c2c4FBE416d9e8655264779e894B6121630` and the code will infer if the contract is the old one or the new one and apply the corresponding logic.

Reference Imgs:

image

image

image

image

### CODE REVIEW

File: src/components/Dashboard/PendingFunds/PendingFunds.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { Segment } from 'decentraland-ui'
+import { FormattedMessage, FormattedNumber } from 'react-intl'
+import './PendingFunds.css'
```
[the remark this answers is not in the document]
Comment (fzavalia) [reply src/components/Dashboard/PendingFunds/PendingFunds.js:4]: Good catch!

[the remark this answers is not in the document]
Comment (fzavalia) [reply src/components/Dashboard/PendingFunds/PendingFunds.js]: I removed it

**Review by andyesp — APPROVED** [review]

File: src/components/Dashboard/Chart/Chart.js [hunk 2]
```diff
@@ -80,6 +88,48 @@ function getVestingData(start, cliff, duration, total, revokeLog) {
...
+  let vestingData = []
+
+  // TODO: This can probably be optimized to avoid doing the same calculations for every day.
+  for (let i = 0; i < vestingDays; i++) {
+    let vestedThatDay
+
+    if (i <= cliffEndDay) {
+      vestedThatDay = 0
+    } else {
+      const elapsedPeriods = (DAY_IN_SECONDS * (i + 1)) / periodDuration
```
Comment (1emu) [inline src/components/Dashboard/Chart/Chart.js:111]: we could add a check for `periodDuration > 0`

File: src/modules/api.js [hunk 3]
```diff
@@ -48,27 +49,46 @@ export default class API {
...
+      await vesting.methods.getIsLinear().call()
+      version = 'v2'
+    } catch (e) {
+      vesting = new eth.Contract(vestingAbi, address)
+      version = 'v1'
+    }
+
+    let tokenContractAddress
+
+    if (version === 'v1') {
```
Comment (1emu) [inline src/modules/api.js:85]: Versions could be extracted to constants to avoid repetition

**Review by 1emu — APPROVED** [review]
✨  LGTM, left some minor suggestions

File: src/components/Dashboard/Chart/Chart.js (hunk 2 above)
Comment (fzavalia) [reply src/components/Dashboard/Chart/Chart.js:115]: Addressed in https://github.com/decentraland/vesting-dashboard/pull/45/commits/f9d176b781f7d5719c403352968a83cd1443fe52

File: src/modules/api.js (hunk 3 above)
Comment (fzavalia) [reply src/modules/api.js]: Addressed in https://github.com/decentraland/vesting-dashboard/pull/45/commits/054b1dfffc8d1d2c57a20cef6babadfc8ab2a996

**Review by andyesp — APPROVED** [review]

**Pull request merged**